### ISSUE TITLE
Add `soft_arg_min` and `soft_arg_max`

### ISSUE THREAD

**Issue by bvssvni** [advancedresearch/poi#984]
https://pytorch.org/docs/stable/nn.functional.html?highlight=softmax#torch.nn.functional.softmax
